src/nvim/eval.c

```diff
@@ -14456,11 +14456,11 @@ static void f_setreg(typval_T *argvars, typval_T *rettv)
   char_u      *strregname;
   char_u      *stropt;
   bool append = false;
-  char_u yank_type;
+  MotionType yank_type;
   long block_len;
 
   block_len = -1;
-  yank_type = MAUTO;
+  yank_type = kUnknownMotion;
```

Comment:
It is better to ask people which add enums that do not follow it why. I do not add such enums. Common prefix indicates that value belongs to some enum, it also provides a way to easily search for all places where any enum value is mentioned.